### DISCUSSION TITLE
Support for Ring Access Controller Pro 2?

### DISCUSSION THREAD

**Discussion by scottjg** [tsightler/ring-mqtt#417] [Feature Requests]
Ring sells an access control module that can trigger automatic gates or magnetic locks: https://ring.com/products/access-controller-pro-2

would it be possible to add support for it? happy to provide any debug information if that might help. In the ring app, it shows up as just another lock.

**Comment by tsightler** [comment]
I don't believe it's possible.  As far as I understand this device is actually an Amazon Key device, and the Ring app actually integrates it by using the Amazon Key API directly, so none of the functions are actually available via the standard Ring API.  To integrate with it someone would need to reverse engineer the Amazon Key API I'd guess.  More details about the original investigation involving the Access Controller Pro are here:

https://github.com/dgreif/ring/issues/384#issuecomment-660169540

**Comment by scottjg** [comment]
interesting, thanks for the info- i was fooled since it showed up alongside my z-wave lock and seemed to work the same way.

**Comment by scottjg** [comment]
in case anyone else comes back to this at some point, it looks like the integration is somehow even more obfuscated than through Amazon Key which is mentioned in the linked issue. my controller was setup through the Ring app (you have to call to activate it), and I didn't want to enable Amazon Key. I found that it doesn't show up in the `ring-device-data-cli` or on the Ring web app - so i think it might be a little annoying to reverse engineer.

**Reply by scottjg** [reply]
in case anyone ever comes back to this, i ended up wiring a 20$ Shelly relay along with the Ring Access Control Pro 2 module. Easier than trying to reverse engineer a non-public API, and probably more reliable. 

The other thing I discovered was that the Ring module only pulses the lock for about one second (not configurable), so i think either way you need something in between the Ring module and the gate. In my case, i have it wired so that the Ring module triggers the Shelly relay, which has a configurable timer. The Shelly relay is already supported by home assistant.
